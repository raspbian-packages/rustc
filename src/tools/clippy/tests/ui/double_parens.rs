#![feature(plugin)]
#![plugin(clippy)]

#![warn(double_parens)]
#![allow(dead_code)]

fn dummy_fn<T>(_: T) {}

struct DummyStruct;

impl DummyStruct {
    fn dummy_method<T>(self, _: T) {}
}

fn simple_double_parens() -> i32 {
    ((0))
}

fn fn_double_parens() {
    dummy_fn((0));
}

fn method_double_parens(x: DummyStruct) {
    x.dummy_method((0));
}

fn tuple_double_parens() -> (i32, i32) {
    ((1, 2))
}

fn unit_double_parens() {
    (())
}

fn fn_tuple_ok() {
    dummy_fn((1, 2));
}

fn method_tuple_ok(x: DummyStruct) {
    x.dummy_method((1, 2));
}

fn fn_unit_ok() {
    dummy_fn(());
}

fn method_unit_ok(x: DummyStruct) {
    x.dummy_method(());
}

fn main() {}
