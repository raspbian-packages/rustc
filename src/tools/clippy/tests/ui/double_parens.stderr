error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:16:5
   |
16 |     ((0))
   |     ^^^^^
   |
   = note: `-D double-parens` implied by `-D warnings`

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:20:14
   |
20 |     dummy_fn((0));
   |              ^^^

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:24:20
   |
24 |     x.dummy_method((0));
   |                    ^^^

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:28:5
   |
28 |     ((1, 2))
   |     ^^^^^^^^

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:32:5
   |
32 |     (())
   |     ^^^^

error: aborting due to 5 previous errors

